### PR TITLE
sof-tplgreader.py: document boolean precedence and other design issues

### PR DESCRIPTION
**Pull request by marc-hb**
Also remove unused imports and add pydocs about data structures. No
functional change.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
No new pylint warning in https://github.com/thesofproject/sof-test/pull/714/checks?check_run_id=2877298503

**Review by fredoh9 — DISMISSED** [review]
Thank you for adding detailed document and comments!

**Comment by marc-hb** [comment]
Thanks @fredoh9 ! For this one I want to for @aiChaoSONG 's review too, he wrote bits of this code.

**Review by aiChaoSONG — APPROVED** [review]
Thanks

File: tools/sof-tplgreader.py [hunk 1]
```diff
@@ -285,12 +305,21 @@ def parse_and_set_block_keyword(block_str, tplgreader):
...
 `-f "type:any` -> all pipelines
 `-f "type:playback"` -> playback pipelines
 `-f "type:capture & pga"` -> capture pipelines with PGA
 `-f "pga & eq"` -> pipelines with both EQ and PGA
 `-f "id:3"` -> pipeline whose id is 3
+
+WARNING: usual boolean precedence does NOT apply! Filters are naively
+applied from left to right. For instance you could expect "id:2 | pga & id:1"
+to be interpreted as "id:2 | (pga & id:1)" and to return nothing.
+"id" cannot be both 1 and 2. But no: this script applies it from
```
Comment (marc-hb) [inline tools/sof-tplgreader.py]: I screwed up a small bit of this explanation, just pushed a fixed version now.

**Review by marc-hb — COMMENTED** [review]
Last minute two-lines long `--help` fix, need your approval again. Sorry for the inconvenience.

**Review by aiChaoSONG — APPROVED** [review]

**Comment by marc-hb** [comment]
2 approvers and pure doc update: can be fixed incrementally later if something is found wrong. Merging.

**Pull request merged**